flytekit/clis/sdk_in_container/run.py

```diff
@@ -130,14 +132,23 @@ def run(
         )
         _, version = script_mode.hash_file(filename)
         remote = FlyteRemote(Config.auto(), default_project=project, default_domain=domain)
-        wf = remote.register_script(
-            wf_entity,
-            project=project,
-            domain=domain,
-            image_config=image_config,
-            destination_dir=destination_dir,
-            version=version,
-        )
+
+        try:
+            wf = remote.fetch_workflow(
+                project=project,
+                domain=domain,
+                name=wf_entity.name,
+                version=version,
+            )
+        except FlyteEntityNotExistException:
```

Comment:
Why do we need this? Why not catch already exists

Reply:
doing it the other way around would produce a `_InactiveRpcError`
```
    raise _InactiveRpcError(state)
grpc._channel._InactiveRpcError: <_InactiveRpcError of RPC that terminated with:
	status = StatusCode.INVALID_ARGUMENT
	details = "task with different structure already exists with id resource_type:TASK project:"flytesnacks" domain:"development" name:"testing_list_pyflyte_run.add_data" version:"e264ed653a92029fe59b53ac62fd9510""
	debug_error_string = "{"created":"@1649691831.919710000","description":"Error received from peer ipv6:[2606:4700:3036::6815:2f8d]:443","file":"src/core/lib/surface/call.cc","file_line":904,"grpc_message":"task with different structure already exists with id resource_type:TASK project:"flytesnacks" domain:"development" name:"testing_list_pyflyte_run.add_data" version:"e264ed653a92029fe59b53ac62fd9510"","grpc_status":3}"
>
```

Unless I'm missing something, this way is cleaner since a `FlyteEntityNotExistException` is actually raised and can be cleanly caught in the `except` clause.